src/LibraryUtilities.ts

```diff
@@ -508,6 +511,37 @@ export function buildLibrarySectionsFromLayoutSpecs(loadedTypes: any, layoutSpec
     return sections;
 }
 
+/**
+ * Update the Category Groups to CoreGroup.
+ * This is to achieve the specific design for librarie.
+ * @param elements
+ */
+function updateCategoryGroups(elements: LayoutElement[]) {
+    if(!elements || elements.length == 0) return;
+
+    //filter the section elemens
+    let sectionElements : LayoutElement[] = elements.filter((elem: LayoutElement) => {
+        return elem.elementType == "section";
+    });
+
+    sectionElements.forEach((section: any) => {
+        //get all the category elements
+        let categoryElements = section.childElements.filter((child: LayoutElement) => {
+            return child.elementType == "category";
+        }); 
+
+        //get the first level group elements
+        let groupElements = categoryElements.filter((child: LayoutElement) => {
+            //change the element type of group elements
+            child.childElements.forEach((grp: LayoutElement) => {
+                if(grp.elementType != "class") {
+                    grp.elementType = "coregroup";
+                }
```

Comment:
I could have used underscore, but I find this way more explicit for someone to understand what we are doing (underscore does this implicit).

Reply:
cool - looks good, can you add just a bit more description to the summary of the behavior - something like: changes categories topLevel children `itemType` from groups to coreGroups.... or something like that.